Unable to use import androidx.appcompat.widget.Toolbar into a java file when using AndroidJavaSource

### Android application type

.NET Android (net7.0-android, etc.)

### Affected platform version

VS 2022 Version 17.6.5

### Description

In my Android .NET 7 app, I have a java file where I'm trying to import `Toolbar` from `androidx.appcompat.widget`:

```
package androidapp5;

import androidx.appcompat.widget.Toolbar;

```
But it gives the following compilation error:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	JAVAC0000	 error: package androidx.appcompat.widget does not exist
import androidx.appcompat.widget.Toolbar;
```

This is the csproj:

```
  <Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net7.0-android</TargetFramework>
    <SupportedOSPlatformVersion>21</SupportedOSPlatformVersion>
    <OutputType>Exe</OutputType>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
    <ApplicationId>com.companyname.AndroidApp5</ApplicationId>
    <ApplicationVersion>1</ApplicationVersion>
    <ApplicationDisplayVersion>1.0</ApplicationDisplayVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Xamarin.AndroidX.AppCompat" Version="1.6.1.3" />
  </ItemGroup>
</Project>
```
I also tried to manually add the following in the csproj but it didn't make any difference:

```
  <ItemGroup>
    <AndroidJavaSource Include="Interop.java" />
  </ItemGroup>
```

Note that when using the Toolbar class in C# code, it compiles OK:

```
using Toolbar = AndroidX.AppCompat.Widget.Toolbar;

namespace AndroidApp5
{
    [Activity(Label = "@string/app_name", MainLauncher = true)]
    public class MainActivity : Activity
    {
        protected override void OnCreate(Bundle? savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            // Set our view from the "main" layout resource
            SetContentView(Resource.Layout.activity_main);

            Toolbar toolbar;
        }
    }
}
```

### Steps to Reproduce

Just create a Android .NET 7 app, add a file called Interop.java like above.

Here's the sample: 

[AndroidApp5.zip](https://github.com/xamarin/xamarin-android/files/12063900/AndroidApp5.zip)

### Did you find any workaround?

_No response_

### Relevant log output


_No response_

MacOSX

`net8.0`

```
dotnet build AndroidApp5/AndroidApp5.csproj /bl
```

output:

```
MSBuild version 17.7.0-preview-23251-02+59879b095 for .NET
  Determining projects to restore...
  Restored /Users/moljac/Downloads/12063900/AndroidApp5/AndroidApp5.csproj (in 686 ms).
/usr/local/share/dotnet/sdk/8.0.100-preview.4.23260.5/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(287,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/Users/moljac/Downloads/12063900/AndroidApp5/AndroidApp5.csproj]
/Users/moljac/Downloads/12063900/AndroidApp5/MainActivity.cs(15,21): warning CS0168: The variable 'toolbar' is declared but never used [/Users/moljac/Downloads/12063900/AndroidApp5/AndroidApp5.csproj]
  AndroidApp5 -> /Users/moljac/Downloads/12063900/AndroidApp5/bin/Debug/net7.0-android/AndroidApp5.dll

Build succeeded.

/Users/moljac/Downloads/12063900/AndroidApp5/MainActivity.cs(15,21): warning CS0168: The variable 'toolbar' is declared but never used [/Users/moljac/Downloads/12063900/AndroidApp5/AndroidApp5.csproj]
    1 Warning(s)
    0 Error(s)

Time Elapsed 00:00:53.47
```

[msbuild.binlog.zip](https://github.com/xamarin/xamarin-android/files/12064211/msbuild.binlog.zip)

--- a/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.BuildOrder.targets
+++ b/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.BuildOrder.targets
@@ -187,6 +187,7 @@ properties that determine build ordering.
     </_ResolveLibraryProjectsDependsOn>
     <_CompileBindingJavaDependsOnTargets>
       _AdjustJavacVersionArguments;
+      _GetLibraryImports;
       _DetermineBindingJavaLibrariesToCompile;
       _GetJavaPlatformJar;
     </_CompileBindingJavaDependsOnTargets>

--- a/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs
@@ -1555,6 +1555,35 @@ namespace UnnamedProject
 		}
 
 		[Test]
+		public void BuildApplicationWithJavaSourceUsingAndroidX ([Values(true, false)] bool isRelease)
+		{
+			var proj = new XamarinAndroidApplicationProject () {
+				IsRelease = isRelease,
+				OtherBuildItems = {
+					new BuildItem (AndroidBuildActions.AndroidJavaSource, "ToolbarEx.java") {
+						TextContent = () => @"package com.unnamedproject.unnamedproject;
+import android.content.Context;
+import androidx.appcompat.widget.Toolbar;
+public class ToolbarEx {
+	public static Toolbar GetToolbar (Context context) {
+		return new Toolbar (context);
+	}
+}
+",
+						Encoding = Encoding.ASCII
+					},
+				}
+			};
+			proj.PackageReferences.Add (KnownPackages.AndroidXAppCompat);
+			using (var b = CreateApkBuilder ()) {
+				b.ThrowOnBuildFailure = false;
+				Assert.IsTrue (b.Build (proj), "Build should have succeeded");
+
+				Assert.IsTrue (b.Clean (proj), "Clean should have succeeded.");
+			}
+		}
+
+		[Test]
 		public void BuildApplicationCheckThatAddStaticResourcesTargetDoesNotRerun ()
 		{
 			var proj = new XamarinAndroidApplicationProject ();

